### PR TITLE
add telemetry and fixing local variable issue

### PR DESCRIPTION
**Pull request by Erli-ms**

### CODE REVIEW

**Review by SamuelBishop — APPROVED** [review]

File: customer-service/omnichannel/real-time-translation/webResourceV2.js [hunk 1]
```diff
@@ -274,50 +295,69 @@ var C1WebResourceNamespace = {
...
 		};
-		translateToC1orC2 = translationConfig.translateToC1orC2;
+		var translateToC1orC2 = translationConfig.translateToC1orC2;
 		// While translating for C1 as we want to use auto detection of langauge for the incoming message.
-		// so, we are not setting the sourceLanguage below.
+		// We should set the sourceLang if we already known the finalC2Language to avoid message change in the middle of the conversation
+		// the drawback for this approach is, if the detection of the language is wrong in the beginning, there is no way to correct it afterwards.
 		if (translateToC1orC2 == Microsoft.Omnichannel.TranslationFramework.TranslateTo.C1) {
-			destLang = C1WebResourceNamespace.dictForAllConversation[conversationId]['C1Lang'];
+			sourceLang = C1WebResourceNamespace.dictForAllConversation[conversationId]['finalC2Lang'];
```
Comment (Erli-ms) [inline customer-service/omnichannel/real-time-translation/webResourceV2.js:339]: to be fixed. we need destLang, mistakenly comment out it.

File: customer-service/omnichannel/real-time-translation/webResourceV2.js [hunk 2]
```diff
@@ -274,50 +297,69 @@ var C1WebResourceNamespace = {
...
 		};
-		translateToC1orC2 = translationConfig.translateToC1orC2;
+		var translateToC1orC2 = translationConfig.translateToC1orC2;
 		// While translating for C1 as we want to use auto detection of langauge for the incoming message.
-		// so, we are not setting the sourceLanguage below.
+		// We should set the sourceLang if we already known the finalC2Language to avoid message change in the middle of the conversation
+		// the drawback for this approach is, if the detection of the language is wrong in the beginning, there is no way to correct it afterwards.
 		if (translateToC1orC2 == Microsoft.Omnichannel.TranslationFramework.TranslateTo.C1) {
-			destLang = C1WebResourceNamespace.dictForAllConversation[conversationId]['C1Lang'];
+			//sourceLang = C1WebResourceNamespace.dictForAllConversation[conversationId]['finalC2Lang']; //to be discussed
```
Comment (Erli-ms) [inline customer-service/omnichannel/real-time-translation/webResourceV2.js:341]: to be removed

File: customer-service/omnichannel/real-time-translation/webResourceV2.js [hunk 3]
```diff
@@ -18,6 +18,8 @@ var C1WebResourceNamespace = {
 	bingTranslateApiClientSecret: '<please add your own azure translation api key>',
 	googleTranslateApiClientSecret: '<please add your own google translation v2 api key>',
 	useAzureTranslationApis: true,//please override it to false if planning to use google translation v2 api
+	messageBuffer: new Map(),
+	enableLanguageDetectionWithHistoryMessages: true,
```
Comment (Erli-ms) [inline customer-service/omnichannel/real-time-translation/webResourceV2.js:22]: Need to change it to false

**Review by karthsMicrosoft — APPROVED** [review]

File: customer-service/omnichannel/real-time-translation/webResourceV2.js [hunk 4]
```diff
@@ -454,21 +531,121 @@ var C1WebResourceNamespace = {
...
+			messageBufferForConversation.shift();
+		}
+		// interface detectedLanguage = {
+		// 	language: string;
+		// 	score: number;
+		// }
+		if (translationResult.detectedLanguage) {
+			messageBufferForConversation.push(translationResult.detectedLanguage);
+		}
+		let messageScoreMap = new Map();
```
[the remark this answers is not in the document]
Comment (Erli-ms) [reply customer-service/omnichannel/real-time-translation/webResourceV2.js:613]: renamed

**Review by shubhadaj — APPROVED** [review]

**Pull request merged**